loopchain/baseservice/peer_manager.py

```diff
@@ -76,59 +70,62 @@ def load(peer_list_data_dumped: bytes):
+        except ValueError:
+            util.logger.warning(f"peer_id({peer_id}) not in peer_list")
+            return None
 
-class PeerManager:
-    def __init__(self, channel_name):
-        """DB에서 기존에 생성된 PeerList 를 가져온다.
-        이때 peer status 는 unknown 으로 리셋한다.
+    def last_peer(self) -> Optional[Peer]:
+        """get last peer in peer_list
+
+        :return:
         """
-        self.peer_list_data = PeerListData()
-        self.__channel_name = channel_name
+        if len(self.peer_list) > 0:
+            return list(self.peer_list.values())[-1]
+        return None
+
+
+class PeerManager:
+    def __init__(self):
+        """Manage peer list in operation."""
+        self._peer_list_data = PeerListData()
 
         # lock object for if add new peer don't have order that must locking
         self.__add_peer_lock: threading.Lock = threading.Lock()
 
     @property
     def peer_list(self) -> dict:
-        """
+        """return peer_list of peer_list_data
 
-        :return: { peer_id:PeerInfo }
+        :return:
         """
-        return self.peer_list_data.peer_list
+        return self._peer_list_data.peer_list
 
     @property
-    def peer_order_list(self) -> dict:
-        """
+    def leader_id(self) -> Optional[str]:
+        """return leader's peer id
 
-        :return: { order:peer_id }
+        :return:
         """
-        return self.peer_list_data.peer_order_list
+        return self._peer_list_data.leader_id
 
-    def peer_ids_hash(self):
-        """ It's temporary develop for Prep test net. This value will replace with Prep root hash.
+    def rep_hash(self) -> str:
+        """return reps root hash as a string.
 
         :return:
         """
-        order_list = list(self.peer_order_list.keys())
-        order_list.sort()
-        peer_count = len(order_list)
-        peer_ids = ""
-
-        for i in range(peer_count):
-            peer_order = order_list[i]
-            peer_id = self.peer_order_list[peer_order]
-            peer_each = self.peer_list[peer_id]
+        block_prover = BlockProver((ExternalAddress.fromhex_address(peer.peer_id).extend()
+                                    for peer in self._peer_list_data.peer_list.values()),
+                                   BlockProverType.Rep)
+        return block_prover.get_proof_root().hex_0x()
 
-            util.logger.debug(f"peer_order({peer_order}), peer_id({peer_id}), peer_target({peer_each.target})")
-            peer_ids += peer_id
-
-        return self.get_peer_ids_hash(peer_ids)
-
-    @staticmethod
-    def get_peer_ids_hash(peer_ids):
-        peer_ids_hash = hashlib.sha256(peer_ids.encode(encoding='UTF-8')).hexdigest()
-        util.logger.debug(f"peer ids hash({peer_ids_hash})")
-        return peer_ids_hash
+    def serialize_as_preps(self) -> list:
+        return [{'id': peer_id, 'p2pEndpoint': peer.target}
+                for peer_id, peer in self._peer_list_data.peer_list.items()]
```

Comment:
Since `p2pEndpoint` name is from IISS API, shouldn't we use `address` for prep's id as well?

Reply:
`id` would be fine, because `address` is for external API.